### DISCUSSION TITLE
feature to access all templates

### DISCUSSION THREAD

**Discussion by weblinggau** [netflie/whatsapp-cloud-api#49] [Ideas]
for the next update, can there be a feature to access all templates, request templates, and add new numbers. 

I think this will be very helpful

**Comment by aalbarca** [comment]
Hi @weblinggau

Thanks for your suggestion. Can you send me the official documentation about your feature request?

Thanks!

**Comment by weblinggau** [comment]
bro @netflie , here's an example of official documentation for message templates might help
https://developers.facebook.com/docs/whatsapp/business-management-api/message-templates

I think this section to add message templates

`curl -X POST "https://graph.facebook.com/v15.0/{whatsapp-business-account-ID}/message_templates
  ?name={template-name}
  &language=en_US
  &category=TRANSACTIONAL,
  &components=[{
       type:BODY, 
       text:{message-text}
     }, 
     {
       type:HEADER, 
       format:IMAGE, 
       example:{header_handle:[{uploaded-image-file-url}]}
     }],
  &access_token={system-user-access-token}"`

**Reply by daniel89fg** [reply]
This interests me, since I currently do it with my own api just for that.

**Comment by weblinggau** [comment]
to add a new phone number is in this section

https://developers.facebook.com/docs/whatsapp/business-management-api/manage-phone-numbers

**Comment by weblinggau** [comment]
or the whole is in the following documentation.. it will be very helpful if this feature is implemented..
https://developers.facebook.com/docs/whatsapp/business-management-api
